packages/arb-token-bridge-ui/src/hooks/useTransactionHistory.ts

```diff
@@ -473,15 +517,39 @@ const useTransactionHistoryWithoutStatuses = (address: Address | undefined) => {
   }
 }
 
+type UsePartialTransactionHistoryProps = {
+  address: Address | undefined
+  // TODO: look for a solution to this. It's used for now so that useEffect that handles pagination runs only a single instance.
+  runFetcher?: boolean
+  stepName: string
+  fetchFor?: {
+    chains?: ChainPair[]
+  }
+  ready?: boolean
+}
+
+function isReady({
+  previousStep
+}: {
+  previousStep: UsePartialTransactionHistoryResult
+}) {
+  if (previousStep.completed) {
+    return true
+  }
+  return previousStep.transactions.length > 0 && !previousStep.loading
+}
+
 /**
  * Maps additional info to previously fetches transaction history, starting with the earliest data.
  * This is done in small batches to safely meet RPC limits.
  */
-export const useTransactionHistory = (
-  address: Address | undefined,
-  // TODO: look for a solution to this. It's used for now so that useEffect that handles pagination runs only a single instance.
-  { runFetcher = false } = {}
-): UseTransactionHistoryResult => {
+const usePartialTransactionHistory = (
```

Comment:
The old `useTransactionHistory` is `usePartialTransactionHistory` now, and the new `useTransactionHistory` returns aggregated results from multiple `usePartialTransactionHistory`.